pkg/destroy/azure/azure.go

```diff
@@ -111,6 +111,12 @@ func (o *ClusterUninstaller) Run() error {
 	waitCtx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
+	// Fail if we can't list anything - This will catch invalid credential errors
+	_, err = o.zonesClient.ListByResourceGroup(waitCtx, o.ResourceGroupName, to.Int32Ptr(1))
+	if err != nil {
+		return err
+	}
+
```

Comment:
So is our goal to exit early on permission issues right?

So what if the user have permission to list the zones but doesn't really have permission to delete the resource group for example, we will again be stuck looping right?

should we handle permisssion denied errors from API calls as early exist ? 
for example, check the error returned from deletePublicRecords and check if the error was permission denied one and if yes exit early from destroy?

Reply:
https://blog.golang.org/go1.13-errors

has information and example on how to assert the error ..